ServantPersistent/src/DBTypes.hs

```diff
@@ -0,0 +1,63 @@
+{-# LANGUAGE DeriveGeneric   #-}
+module DBTypes where
+
+import Data.Aeson
+import Data.Aeson.TH
+import Data.Text
+import GHC.Generics
+import Control.Lens
+import Types
+import Models
+
+type TenantID = Key DBTenant
+type Tenant = DBTenant
+type TenantOutput = DBTenant
+type UserID = Key DBUser
+
+data DBError = TenantNotFound TenantID
+             | UserNotFound UserID deriving (Eq, Show)
+data UserCreationError = UserExists Text
+                       | TenantDoesn'tExist Text
+                        deriving (Eq, Show)
+
+data Role = SiteAdmin
```

Comment:
the roles are not supposed to be hard-coded. Please look at the relevant [tables in the schema](https://github.com/vacationlabs/haskell-webapps/blob/366cce31be642604077397bdcda966fa22c43686/ServantOpaleye/db/schema.sql#L59-L81) to understand that roles are defined by the tenant, as **any collection** of pre-defined permissions.